sphinx/writers/html5.py

```diff
@@ -420,7 +420,7 @@ def depart_caption(self, node):
 
     def visit_doctest_block(self, node):
         # type: (nodes.doctest_block) -> None
-        self.visit_literal_block(node)
+        self.visit_literal_block(node)  # type: ignore
```

Comment:
I feel currrent annotations for visitor methods are too strict. I understand it is better to use concrete types to represent what type of node visitor methods accept. But it is too sensitive for authors of derived implementations.
Personally, using `nodes.Element` would better for writers. It is ambiguous, but easy to use.

@cocoatomo what do you think?